### ISSUE TITLE
performance.measure() is crashing my route

### ISSUE THREAD

**Issue by devin-liu** [vercel/next.js#20743]
**What version of Next.js are you using?**

10.0.4

**What version of Node.js are you using?**

v14.4.0

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**How are you deploying your application?**

Vercel

**Describe the Bug**

I'm building a google oauth flow. I get the oauth menu account list opened, and click on my google account. It redirects back - unsuccessfully, to my app page. I get these errors

DOMException: Failed to execute 'measure' on 'Performance': The mark 'beforeRender' does not exist.

Uncaught (in promise) Error: Cancel rendering route

**Expected Behavior**

I'm building a google oauth flow. I get the oauth menu account list opened, and click on my google account. It redirects back - successfully, to my app page.

**To Reproduce**

I believe the error comes from this line. I suspect because my page is redirecting from google back to my app, the mark is not there or was cleared out. 

https://github.com/vercel/next.js/blob/dbe1e626f877a9ef7bbffc2f360bae809874790d/packages/next/client/index.tsx#L534

**Comment by tarunama** [comment]
Can reproduce error with below example ?
https://github.com/vercel/next.js/tree/canary/examples/with-next-auth

I tried to reproduce your error, but error did not occur.
https://github.com/tarunama/with-next-google-oauth

**Comment by bwillem** [comment]
Looks like a separate bug? But I had performance reporting crash an API route in production as well - here's the Sentry stack trace

<img width="798" alt="Screen Shot 2021-04-28 at 10 57 54 AM" src="https://user-images.githubusercontent.com/5076181/116450950-c2701300-a810-11eb-901f-9e5071baece9.png">

**Comment by threehams** [comment]
I have a non-minimal repro case for this error, though under completely different circumstances, here:
https://github.com/threehams/barrys-time-adventures-nx/tree/nextjs-perf-bug

Windows 10
Node 16.9.1
npm 7.21.1

Requires `npm install --legacy-peer-deps`, which is likely part of the problem.
Run the server with `npx nx serve barry`. Happens immediately on server start.

**Comment by vitalybaev** [comment]
We've also experienced such problem on `12.1.6`. I'll try to get more details here.

<img width="854" alt="CleanShot 2022-05-25 at 13 19 08@2x" src="https://user-images.githubusercontent.com/724423/170228338-cfc006bf-43a2-4baf-8ada-6f4e2123941a.png">

**Comment by sairion** [comment]
I'm seeing this as well on production. It's not very frequent, but a few users are seeing this on a random basis.
it would be nice if this doesn't break the route, at least.

**Comment by vedraan** [comment]
I'm also seeing this in production but not too often. It seems to be happening if a browser tab with the app is left open but inactive, and users come back to it after a while.

`Failed to execute 'measure' on 'Performance': The mark 'beforeRender' does not exist.`

Next version `12.1.6`

**Comment by m1guelsb** [comment]
In my case, looks like using useCallback + useEffect hook inside a Context is what was causing that error, when I rewrote the code, the error sttoped. Very strange.

**Comment by CRIMSON-CORP** [comment]
I'm my case, i was getting the when i use router.push() to the page i was currently on

**Comment by lveillard** [comment]
Same as @CRIMSON-CORP 
Had to compare with `router.pathname` to avoid doing the route.push to the same route

I think it could be nice to create a dedicated error message for this case

**Comment by abdulsalamcodes** [comment]
I got the error also because I was using `router.push` to push to the same page as the page I am currently on.

**Comment by devramsean0** [comment]
I'm also getting this error, but not when using router.push, using `location.href` instead and i'm still getting the error

**Comment by kingsley-sunny** [comment]
> I'm also seeing this in production but not too often. It seems to be happening if a browser tab with the app is left open but inactive, and users come back to it after a wh

In my case router.replace() to the page that i was already on was causing that error

**Comment by michaelangrivera** [comment]
@balazsorban44 
@lluia 
can we get some visibility on this? I'm getting the same error as well. 
This issue was opened in 2021.

**Comment by douglasjunior** [comment]
I`m getting this error when using [MSAL.js](https://github.com/AzureAD/microsoft-authentication-library-for-js) in a mobile browser.

The error happens at some point when it is redirected to the Microsoft login page, and then returns as a callback to my Next.js page.

**Comment by rangshuman** [comment]
> I'm my case, i was getting the error when i use router.push() to the page i was currently on

This was exactly what I was doing in my case. Pushing to the same router I was currently on created this issue for me. Thanks for this comment @CRIMSON-CORP

**Comment by JuniorTour** [comment]
Personally I think, this issue can be closed.
1. I made some investigation based on this reproduction: https://github.com/threehams/barrys-time-adventures-nx/tree/nextjs-perf-bug

It is a usage mistake, not a bug of next.js.

2. Another router.push() erro: 
I cant reproduce it, but according to comments bellow, it is also usage mistake, not a bug of next.js.

So I think, this issue can be closed.

**Comment by capaj** [comment]
Still getting it in next@13.1.2

would be nice to get this fixed.

**Comment by Anvarmirzo** [comment]
in my case, this error is triggered when using router.push twice in different components, in one component I collected all query parameters and added a new parameter with them, in the other I also collected all query parameters and added a new parameter with them

**Comment by desnor** [comment]
We are also encountering this issue occasionally when a page has been open for a while and focus has been on a different browser tab. When the user returns it goes into a sort of loop with the error message `DOMException: Failed to execute 'measure' on 'Performance': The mark 'beforeRender' does not exist.`

This is happening after the static export of the app, not using any calls to `router.push`. We are using the query params from the `router` only. 

`performance.measure("Next.js-route-change-to-render", e[0].name, "beforeRender"),` is the source line causing this, in the minified output of the chunks `main` next files.

It seems like the app is trying to reroute the page to the same route - this warning pops up in the dev console in Chrome

```
main-627138c22ebf06ea.js:1 Throttling navigation to prevent the browser from hanging. See https://crbug.com/1038223. Command line switch --disable-ipc-flooding-protection can be used to bypass the protection 
```

Not sure if that adds anything useful for someone?

**Comment by itsjavi** [comment]
In my case this occurs when I use an external component in `_document.tsx` 's Head. My component comes from my component library and it looks like this:

```AssetsHead.tsx
  // ...
  return (
    <>
      <link id="AppIcons" rel="stylesheet" href={`${assetsUrl}/dist/fonts/AppIcons/AppIcons.css`} />
      <link
        id="GlyphIcons"
        rel="stylesheet"
        href={`${assetsUrl}/dist/fonts/PokeGlyphs/GlyphIcons.css`}
      />
    </>
  )
```

So my workaround was to use it inside a `next/head` in my app layout, even though nextjs will complain with warnings to not import stylesheets like this.

**Comment by mustafa-hanif** [comment]
> In my case this triggered intermittently when I refreshed the page, when I accidentally used a `next/document` `Head` inside another. (NextJS 13.2)
> 
> I think NextJS should detect this case and report a proper error.

This is exactly what my issue was, thanks!

**Comment by ClemCa** [comment]
Had the same with replace, when the page stayed the same as well.

**Comment by shangcode** [comment]
Same error,  nextjs version 13.1.6 with next-auth 4.20.1😿

**Comment by am1rb** [comment]
I faced the `Failed to execute 'measure' on 'Performance': The mark 'beforeRender' does not exist.` error in next.js `13.1.6`. It happens when someone tries to load a page and, before it gets fully loaded, clicks on the browser back/forward button.

You can see the steps to reproduce it here: https://github.com/vercel/next.js/issues/47560#issuecomment-1488426370

**Comment by zodman** [comment]
> We are also encountering this issue occasionally when a page has been open for a while and focus has been on a different browser tab. When the user returns it goes into a sort of loop with the error message `DOMException: Failed to execute 'measure' on 'Performance': The mark 'beforeRender' does not exist.`
> 
> This is happening after the static export of the app, not using any calls to `router` at all, on a page that renders only plain HTML.
> 
> `performance.measure("Next.js-route-change-to-render", e[0].name, "beforeRender"),` is the source line causing this, in the minified output of the chunks `main` next files.
> 
> It seems like the app is trying to reroute the page to the same route - this warning pops up in the dev console in Chrome
> 
> ```
> main-627138c22ebf06ea.js:1 Throttling navigation to prevent the browser from hanging. See https://crbug.com/1038223. Command line switch --disable-ipc-flooding-protection can be used to bypass the protection 
> ```
> 
> Not sure if that adds anything useful for someone?

I think i have the same issue .. @desnor  what version of nextjs you have ?

**Comment by shawngustaw** [comment]
I'm experiencing this with `router.replace` as well when the `pathname` is the same as the current url. I'm basically just using it to update the query parameters for the current route.

**Comment by shawngustaw** [comment]
It seems to go away if you add `{ shallow: true }` to `router.replace`. I still get this error popping up but it doesn't break the page: `Error: Loading initial props cancelled`

**Comment by dellwatson** [comment]
i dont even have router push and this crashing my next app too

**Comment by gimenete** [comment]
Happened to me for having two `<Head>`s in the same component.

**Comment by rogeliorv** [comment]
Can replicate with router.replace

**Comment by rwieruch** [comment]
Same issue I think: https://github.com/vercel/next.js/issues/47560#issuecomment-1661807784

**Comment by rwieruch** [comment]
More details on this bug. I get it whenever the pathname stays the same but the queries change. For instance, when the current `pathname` is

```
/cars/6/?selectedCarId=1
```

and I cange it to

```
router.push(
  /cars/6/?selectedCarId=2
);
```

So essentially the asPath stays the same, but the queries change.

**Comment by huozhi** [comment]
The fix is now landed in stable version 13.5.1
